apps/prairielearn/elements/pl-string-input/pl-string-input.mustache

```diff
@@ -1,66 +1,125 @@
+                this.scrollHeight + parseFloat(style.paddingTop) + parseFloat(style.paddingBottom) + 'px';
+            }
+            });
+            input.dispatchEvent(new Event('input'));
+        });
+    });
+</script>
+
 {{#question}}
+
 {{#inline}}<span class="form-inline d-inline-block ml-2">{{/inline}}
-    <span class="input-group pl-string-input">
+    {{#multiline}}
         {{#label}}
-        <span class="input-group-prepend">
-            <span class="input-group-text" id="pl-string-input-{{uuid}}-label">{{{label}}}</span>
-        </span>
+        <label
+            id="pl-string-input-{{uuid}}-label"
+            class="form-label"
+            for="pl-string-input-{{uuid}}-area"
+        >{{{label}}}</label>
         {{/label}}
-        <input
-            name="{{name}}"
-            type="text"
-            inputmode="text"
-            class="form-control pl-string-input-input"
-            size="{{size}}"
-            autocomplete="off"
-            autocorrect="off"
-            {{^editable}}disabled{{/editable}}
-            {{#raw_submitted_answer}}value="{{raw_submitted_answer}}"{{/raw_submitted_answer}}
-            aria-describedby="pl-symbolic-input-{{uuid}}-suffix"
-            aria-labelledby="pl-symbolic-input-{{uuid}}-label"
-            placeholder="{{placeholder}}"
-        />
-        <span class="input-group-append">
-            {{#suffix}}<span class="input-group-text" id="pl-symbolic-input-{{uuid}}-suffix">{{suffix}}</span>{{/suffix}}
-
-            {{#show_info}}
-                <a role="button" class="btn btn-light border d-flex align-items-center" data-toggle="popover" data-html="true" title="String" data-content="{{info}}" data-placement="auto" data-trigger="focus" tabindex="0">
-                    <i class="fa fa-question-circle" aria-hidden="true"></i>
-                </a>
-            {{/show_info}}
+        <span id="pl-string-input-{{uuid}}" class="input-group pl-string-input">
+            <textarea
+                wrap="soft"
+                id="pl-string-input-{{uuid}}-area"
+                name="{{name}}"
+                type="text"
+                inputmode="text"
+                class="form-control pl-string-input-input multiline-input"
+                cols="{{size}}"
+                rows="2"
+                autocomplete="off"
+                autocorrect="off"
+                {{^editable}}disabled{{/editable}}
+                aria-describedby="pl-string-input-{{uuid}}-suffix"
+                placeholder="{{placeholder}}"
+            >{{raw_submitted_answer}}</textarea>
+    {{/multiline}}
+    {{^multiline}}
+        <span id="pl-string-input-{{uuid}}" class="input-group pl-string-input">
+            {{#label}}
+            <span class="input-group-prepend">
+                <span
+                    id="pl-string-input-{{uuid}}-label"
+                    class="input-group-text"
+                    for="pl-string-input-{{uuid}}-input"
```

Comment:
`for` is not a meaningful attribute on a `span`. The previous usage of `aria-labelledby` on the input was correct.

Maybe I haven't been explaining this well, but here's as much relevant information as I think I can share about all this:

- All inputs must have a label.
- If `<label>` can be used, that's the best option. If `<label>` _is_ used, you should use `for` on the label to reference the ID of the labelled element.
- If `<label>` cannot be used, `aria-labelledby` should be used instead. That should refer to the ID of the element serving as the label for the input.

Reply:
Thank you for explaining! That makes sense, I was confused as to why the `aria-labelledby` was being used instead of the `for`.

Reply:
Changed here https://github.com/PrairieLearn/PrairieLearn/pull/9438/commits/034e1035f27db64dbf47102accab254a7e002645